### DISCUSSION TITLE
charlie's garbage github blogpost!

### DISCUSSION THREAD

**Discussion by charlie-sans** [OpenStudioCorp/PythonicOS#29] [General]
Hey! this is my garbage blogposts for updates on PythonicOS and anything related to it or what ever!

so i figured out that you can put commands/functions inside of print()

```python

print(os.getcwd())

```

never new you could do that!

**Comment by charlie-sans** [comment]
### i got some work done when it comes to the desktop display part.

i added a couple things 

1. a folder display that when the script scans the system directory for the system if the resulting file returns a folder, it gets colored light green!
2.  a built in file explorer that i plan to make separate to the main PythonicOS.py script.
3. i converted the time.py script to display in a Tkinter window instead of the zsh/cmd line.
4. added a blue screen function so that if your program calls say 

```python
def pycrashexplorer():
      subprocess.Popen(['python', 'python/pycrashfile.py'])
```
from the main module. your program has to first run
```python 
root.destroy()

#then


subprocess.Popen(['python', 'python/pycrashfile.py'])
``` 

so that when the bluescreen happens the system can kill main.py and PythonicOS.py to restart the system live.

5. new commands for the shell, i am still working on this at the moment so bear with me on that one. 

so with the shell script, that acts like zsh on linux,terminal on mac or the command prompt on windows.
i plan to make or get help with making a terminal like interface so that you can switch from one to another if the python version is not good enough!


thats all that i have for this at the moment, i plan to make more of these later! if you have a question of my methods or anything. just make a  conversation and then refrence #29 and ill come and help! cheers -charlie!

**Comment by charlie-sans** [comment]
I HATE ICONS IN TKINTER!!!!!

i wanted to see if i could add icons to the widgets on the desktop so they wont be a solid color.

when i tried to do that i wouldn't find the path to the DAMN PICTURESSS AHHHHH.

if you want to give it a go, just point the picture path to the system/bin folder in the load_files_thread() functions if ext ==, as that contains every system picture/icon that you could use.

and when you do figure out how to do it PLEASE submit a issue with the changes that you made and ill add them.

**Comment by charlie-sans** [comment]
hey! so i did some things and made a official package manager website! tho it's still being made so i cant share a link as at somepoint it might be broken. 

tho i do live stream it over at https://www.youtube.com/@Charlie_san 

i plan to start doing the low level stuff later on in the week, i might ask h4rldev if they can help with the low level stuff with PythonicOS?

**Comment by charlie-sans** [comment]
around the 17th of June I compiled the Linux kernel to try and make the base layer of PythonicOS, now i need to mount it and edit the files to include a installer, i might modify a mini install of ubuntu for that so yeah.

**Comment by charlie-sans** [comment]
hey hey! i nearky have a completed build of PythonicOS!!!!

**Comment by charlie-sans** [comment]
so. i've added a new build of Pythonic os and now i've created a porgramming language!

**Reply by ProSureString** [reply]
por????
a

**Comment by ghost** [comment]
PythonicOS 🔛🔝

**Reply by ProSureString** [reply]
fr

**Reply by ProSureString** [reply]
facts

**Comment by charlie-sans** [comment]
hey everyone! welcome to the new year!!!!!

i'm fine when it comes to things, i'm taking a break for 2 months from doing and really hard work and working like everyday. 

i've released 1.2.2 of CubeScript and i'm rewriting it for 1.3

unfortunatly 1.2.2 got reported as malware so i've dug up the old src code for 1.2.2 and added that to a lts branch so i can fix it and sign the code when it gets built so peoples computer stops screaming at my horrible code.

**Comment by charlie-sans** [comment]
but on the other hand everything is actually going great except the fact that i have no idea what i want to do with the VR game at this point. I'm getting the user controls setup so that's not what i'm worried about. the snapping part of the game however, where the notes are suppose to snap to a grid and allow them to play a sound is what i'm worried about. my programing knowledge doesn't go that far and i think i might not be able to make it work. i might be able to but we will have to see. when i do manage to get it in working order however, i might host a prerelease on a web host so people can play it ether on desktop or VR???

**Comment by charlie-sans** [comment]
but it's great to be back developing programs again

**Reply by ProSureString** [reply]
fr
